topdown/http.go

```diff
@@ -1277,15 +1351,7 @@ func (c *interQueryCache) CheckCache() (ast.Value, error) {
 		return nil, handleHTTPSendErr(c.bctx, err)
 	}
 
-	resp, err := c.checkHTTPSendInterQueryCache()
```

Comment:
If we enter error responses in the intra query cache either due http errors (eg. connectivity issues) or non-cacheable responses, the fallback mechanism on the intra query cache should still work 🤔

Reply:
It should. I flipped the order, so that we always check the intra-query cache before the inter-query one. Mostly to make it more explicit that we want intra-query consistency above all else, but also because at one point I thought this was going to be necessary.. This change isn't strictly necessary, though. If we make the change I suggested earlier, to always populate the intra-query cache also when inter-query is enabled, then this would be the right thing to do.